### PR TITLE
update health-client to v4 API

### PR DESCRIPTION
**Pull request by jpappa200**
Closes: #7191 
Updates to use traffic ops api v4. Also adds reload capability to systemd

<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Control Health Client (tc-health-client)

## What is the best way to verify this PR?

Build health client and install on admin_down cache. change config file and run systemctl reload tc-health-client and monitor log messages

## If this is a bugfix, which Traffic Control versions contained the bug?

Not a bug fix

## PR submission checklist

- [x] This PR has documentation 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: tc-health-client/tmagent/tmagent.go
Comment (zrhoffman) [inline tc-health-client/tmagent/tmagent.go:661]: The `sv.Type` comparison should be against a shared constant like `tc.MonitorTypeName` rather than the string literal `"RASCAL"`.

https://github.com/apache/trafficcontrol/blob/1c26082ebff759f4311845caee0d5cc1e0a4baa4/lib/go-tc/enum.go#L134-L136

- It's unfortunate that `tc.MonitorTypeName`'s documentation appears to discourages its use. It was added in #6040, and if I had noticed it at the time, I would have suggested rewording it before merging #6040. I think what it's trying to say is that its value could potentially be updated to something else in a future PR, but @ocket8888 can tell us for sure what he was trying to say.

- `tc.MonitorTypeName` changed from `"RASCAL"` to `"TRAFFIC_MONITIOR"` in #7068

File: CHANGELOG.md
Comment (zrhoffman) [inline CHANGELOG.md:27]: Why does #7192 add a changelog entry for #7068?

**Review by zrhoffman — CHANGES_REQUESTED** [review]

File: CHANGELOG.md
Comment (jpappa200) [reply CHANGELOG.md:27]: I did a poor job of resolving the conflict. let me see if I can fix it

File: tc-health-client/tmagent/tmagent.go
Comment (jpappa200) [reply tc-health-client/tmagent/tmagent.go]: Only compared against shared constant.

Comment (zrhoffman) [reply tc-health-client/tmagent/tmagent.go]: I just read the comment (and noticed the rest of the code line, which includes `tc.MonitorTypeName` anyway).

[this will allow it to work either way and can be removed once TO is updated.](https://github.com/apache/trafficcontrol/blob/1c26082ebff759f4311845caee0d5cc1e0a4baa4/tc-health-client/tmagent/tmagent.go#L660)

I agree that checking for both `"TRAFFIC_MONITIOR"` and `"RASCAL"` is useful here, but I still think only shared constants should be used. How about making a new constant for "`RASCAL"`? Not sure how to version it, but there's def value in keeping the old one around to help with upgrades

**Comment by zrhoffman** [comment]
This looks good, but let's wait until #7190 is merged or closed to merge #7192.

File: lib/go-tc/enum.go
Comment (zrhoffman) [inline lib/go-tc/enum.go:154]: Since #7190 revert renaming `RASCAL` to `TRAFFIC_MONITOR`, this part is not needed.

File: tc-health-client/tmagent/tmagent.go
Comment (zrhoffman) [inline tc-health-client/tmagent/tmagent.go:659]: `|| sv.Type == tc.RascalTypeName` is not needed, since #7190 revert renaming `RASCAL` to `TRAFFIC_MONITOR`.

**Review by zrhoffman — CHANGES_REQUESTED** [review]

File: lib/go-tc/enum.go
Comment (jpappa200) [reply lib/go-tc/enum.go]: removed

**Review by zrhoffman — APPROVED** [review]
Awesome!

**Pull request merged**
